src/main.js

```diff
@@ -206,32 +206,32 @@ $(document).ready(() => {
     function setupSidebarFloatingBehaviors() {
       const MOUSE_LEAVE_DELAY = 2000;
       const KEY_PRESS_DELAY = 4000;
-      let isMouseInSidebar = false;
+      const notInSidebar = (event) => !$sidebar.find(event.target).length;
 
       // Opens when mouse is over the handle.
       $toggler.mouseenter(() => toggleSidebar(true));
 
       // Immediately closes if click outside the sidebar.
-      $document.on('click', () => {
-        if (!isMouseInSidebar && !isSidebarPinned() && isSidebarVisible()) {
+      $document.on('click', (event) => {
+        /**
+         * Ensure the event click being triggered outside the sidebar
+         */
+        if (notInSidebar(event) && !isSidebarPinned() && isSidebarVisible()) {
           toggleSidebar(false);
         }
       });
 
       let timerId = null;
-      const startTimer = (delay) => {
-        if (!isMouseInSidebar && !isSidebarPinned()) {
+      const startTimer = (delay, isNotInSidebar) => {
+        if (isNotInSidebar && !isSidebarPinned()) {
           clearTimer();
           timerId = setTimeout(() => toggleSidebar(isSidebarPinned()), delay);
         }
       };
       const clearTimer = () => timerId && clearTimeout(timerId);
 
-      $sidebar.on('keyup', () => startTimer(KEY_PRESS_DELAY));
-      $sidebar.on('mouseleave', () => {
-        isMouseInSidebar = false;
-        startTimer(MOUSE_LEAVE_DELAY);
-      });
+      $sidebar.on('keyup', () => startTimer(KEY_PRESS_DELAY, false));
```

Comment:
Hmm, this doesn't make sense. So it always passes 'false', meaning the timer doesn't start. So what's he point of this entire inline at all?